lib/patch/functions.js

```diff
@@ -48,6 +48,49 @@ function patchSetClearFunction(obj, fnNames) {
   });
 };
 
+
+/**
+ * Patch requestAnimationFrame.  This function is typically recursively called from within
+ * the callback function that it executes.
+ *
+ *  1) If the patched function is called recursively, bind to the
+ *     parent zone and increment the reference count.
+ *  2) After the callback function has executed, decrement the reference count.
+ *  3) If the reference count is <= 0, dequeue the current task.
+ */
+function patchRequestAnimationFrame(obj, fnNames) {
+  fnNames.forEach(function (name) {
+    var delegate = obj[name];
+    if (delegate) {
+      global.zone[name] = function (fn) {
+        if (fn) {
```

Comment:
I adopted your changes where possible, but a few subtleties are lost in translation:
 - The decrement cannot happen before the original callback function is executed.  If it does the count will never be > 0 and any rAF calls inside it will not bind to the parent.  
 - In order to ensure the reference is decremented I would need to wrap the `fn.apply` in a try/catch block.  Are there concerns wrapping each callback?

**important**
I also failed to consider the asynchronous recursive function call cases, i.e. what if you call setTimeout -> requestAnimationFrame from within a rAF callback.  The setTimeout call will maintain the zone, because it is patched, and the new call to rAF will have no reference count, and will incorrectly bind a new zone. 

I constructed a simple test case and it demonstrates that the zones grow ad-infinitum in when used as described. 

I have no time to work on this issue now, but I will take a look at it after work is done for the day.  Here's the spec change if you are interested in looking into it.

```javascript
  it('should bind to parent zone when called recursively in an async fashion', function (done) {
    testZone.run(function () {
      var frames = 0;

      function frameCallback() {
        expect(zone).toBeDirectChildOf(testZone);
        if (frames++ > 15) {
          return done();
        }
        setTimeout(function(){
          rAF(frameCallback);
        },1);
      }

      rAF(frameCallback);
    });
  });

```